src/API/Site/Controllers/Ads/BudgetRecommendationController.php

```diff
@@ -52,47 +52,90 @@ public function __construct( RESTServer $rest_server, BudgetRecommendationQuery
 	 */
 	public function register_routes(): void {
 		$this->register_route(
-			'ads/campaigns/budget-recommendation/(?P<country_code>\\w{2})',
+			'ads/campaigns/budget-recommendation',
 			[
 				[
 					'methods'             => TransportMethods::READABLE,
 					'callback'            => $this->get_budget_recommendation_callback(),
 					'permission_callback' => $this->get_permission_callback(),
+					'args'                => $this->get_collection_params(),
 				],
 				'schema' => $this->get_api_response_schema_callback(),
 			]
 		);
 	}
+
+	/**
+	 * Get the query params for collections.
+	 *
+	 * @return array
+	 */
+	public function get_collection_params(): array {
+		return [
+			'context'       => $this->get_context_param( [ 'default' => 'view' ] ),
+			'country_codes' => [
+				'type'              => 'array',
+				'sanitize_callback' => $this->get_country_code_sanitize_callback(),
+				'validate_callback' => $this->get_country_code_validate_callback(),
+				'items'             => [
+					'type' => 'string',
+				],
+				'required'          => true,
+			],
+		];
+	}
+
 	/**
 	 * @return callable
 	 */
 	protected function get_budget_recommendation_callback(): callable {
 		return function( Request $request ) {
-			$country        = strtoupper( $request->get_param( 'country_code' ) );
-			$currency       = $this->middleware->get_ads_currency();
-			$recommendation = $this
+			$country_codes = $request->get_param( 'country_codes' );
+			$currency      = $this->middleware->get_ads_currency();
+
+			if ( ! $currency ) {
+				return new Response(
+					[
+						'message'       => __( 'No currency available for the Ads account.', 'google-listings-and-ads' ),
+						'currency'      => $currency,
+						'country_codes' => $country_codes,
+					],
+					400
+				);
+			}
+
+			$recommendations = $this
 				->budget_recommendation_query
-				->where( 'country', $country )
+				->where( 'country', $country_codes, 'IN' )
 				->where( 'currency', $currency )
-				->get_row();
+				->get_results();
 
-			if ( ! $recommendation ) {
+			if ( ! $recommendations ) {
 				return new Response(
 					[
-						'message'      => __( 'Invalid country/currency combination', 'google-listings-and-ads' ),
-						'currency'     => $currency,
-						'country_code' => $country,
+						'message'       => __( 'Cannot find any budget recommendations.', 'google-listings-and-ads' ),
+						'currency'      => $currency,
+						'country_codes' => $country_codes,
 					],
-					400
+					404
 				);
 			}
 
+			$returned_recommendations = array_map(
+				function ( $recommendation ) {
+					return [
+						'country'           => $recommendation['country'],
+						'daily_budget_low'  => (float) $recommendation['daily_budget_low'],
+						'daily_budget_high' => (float) $recommendation['daily_budget_high'],
```

Comment:
Do we need these to be cast to floats? In the database the columns are defined as `int(11)` so seems like casting to an int would do here.